crates/core/src/callgrind_flag.rs

```diff
@@ -0,0 +1,61 @@
+use std::hint::black_box;
+use std::sync::atomic::{AtomicU32, Ordering};
+
+/// This module allows temporarily enabling callgrind in async SpacetimeDB code, guarded by a global enable flag.
```

Comment:
I'm fine with this strategy for injecting `flag` into the call stack. I get that it's a little brittle in the sense of needing to align a tiny bit of ABI across to the benchmark crates, but I think it's not that bad.